### PR TITLE
[17_] Implement "Update lots" flow (code refactoring)

### PR DESCRIPTION
**Pull request by IvanShetkoGodel**
Removed:
-redundant changing of types (from Lot to LotDto)
-redundant parameters to functions
-typos in naming

Changed:
-working with DB
-logging messages become more informative

Added:
-check for null argument

### CODE REVIEW

File: GodelMastery.FleaMarket/GodelMastery.FleaMarket.BL/Services/LotService.cs [hunk 1]
```diff
@@ -17,22 +17,23 @@ namespace GodelMastery.FleaMarket.BL.Services
...
             IUnitOfWork unitOfWork, 
             IHtmlParserProvider htmlParserProvider,
             ILotModelFactory lotModelFactory,
             IFilterModelFactory filterModelFactory) : base(unitOfWork)
         {
-            this.unitOfWork = unitOfWork;
-            this.htmlParserProvider = htmlParserProvider;
-            this.lotModelFactory = lotModelFactory;
-            this.filterModelFactory = filterModelFactory;
+            this.unitOfWork = unitOfWork ?? throw new ArgumentNullException(nameof(unitOfWork));
```
Comment (vyshkouski-godeltech-intelliflo) [inline GodelMastery.FleaMarket/GodelMastery.FleaMarket.BL/Services/LotService.cs]: @IvanShetkoGodel As I see, unitOfWork protected field is implemented in BaseService. So I advice you to remove this check from this constructor and paste it to BaseService constructor.

File: GodelMastery.FleaMarket/GodelMastery.FleaMarket.BL/Services/LotService.cs [hunk 2]
```diff
@@ -76,58 +78,66 @@ public IEnumerable<LotDto> GetLotDtos(int filterId)
             catch(Exception e)
             {
                 unitOfWork.RollBack();
-                logger.Error($"Update exception {e.Message}");
+                logger.Error($"Exception occurred in operation UpdateLots for filter with id: {filterId} error message: {e.Message}");
```
Comment (vyshkouski-godeltech-intelliflo) [inline GodelMastery.FleaMarket/GodelMastery.FleaMarket.BL/Services/LotService.cs]: @IvanShetkoGodel Change please to:
Exception occurred during operation UpdateLots for filter with id: {filterId}. Error message:

File: GodelMastery.FleaMarket/GodelMastery.FleaMarket.BL/Services/LotService.cs [hunk 3]
```diff
@@ -76,58 +78,66 @@ public IEnumerable<LotDto> GetLotDtos(int filterId)
...
         }
-        private void UpdatePrice(IEnumerable<Lot> toUpdateLots, Filter currentFilter)
+
+        private void UpdatePrice(IEnumerable<Lot> toUpdateLots, int filterId)
         {
             foreach (var lot in toUpdateLots)
             {
                 logger.Info($"Update lot {lot.Id}");
-                var lotFromDb = currentFilter.Lots.Where(x => x.SourceId == lot.SourceId).FirstOrDefault();
+                var lotFromDb = unitOfWork.Lots.Where(x => x.SourceId == lot.SourceId && x.FilterId == filterId).FirstOrDefault();
```
Comment (vyshkouski-godeltech-intelliflo) [inline GodelMastery.FleaMarket/GodelMastery.FleaMarket.BL/Services/LotService.cs]: @IvanShetkoGodel As toUpdateLots is a collection of existed in db lots, I think you can use here easier way to find them:
unitOfWork.Lots.First(x => x.Id == lot.Id);

File: GodelMastery.FleaMarket/GodelMastery.FleaMarket.BL/Services/LotService.cs [hunk 4]
```diff
@@ -76,58 +78,66 @@ public IEnumerable<LotDto> GetLotDtos(int filterId)
...
         }
-        private void DeleteNotActualLots(IEnumerable<Lot> notActualLot, Filter currentFilter)
+
+        private void DeleteNotActualLots(IEnumerable<Lot> notActualLot, int filterId)
         {
             foreach (var lot in notActualLot)
             {
                 logger.Info($"Delete lot {lot.Id}");
-                var lotFromDb = currentFilter.Lots.Where(x => x.SourceId == lot.SourceId).FirstOrDefault();
+                var lotFromDb = unitOfWork.Lots.Where(x => x.SourceId == lot.SourceId && x.FilterId == filterId).FirstOrDefault();
```
Comment (vyshkouski-godeltech-intelliflo) [inline GodelMastery.FleaMarket/GodelMastery.FleaMarket.BL/Services/LotService.cs]: @IvanShetkoGodel The same

File: GodelMastery.FleaMarket/GodelMastery.FleaMarket.BL/Services/LotService.cs [hunk 5]
```diff
@@ -76,58 +78,66 @@ public IEnumerable<LotDto> GetLotDtos(int filterId)
...
                 unitOfWork.Lots.Delete(lotFromDb);
             }
         }
```
Comment (vyshkouski-godeltech-intelliflo) [inline GodelMastery.FleaMarket/GodelMastery.FleaMarket.BL/Services/LotService.cs]: @IvanShetkoGodel Add empty line here

**Comment by IvanShetkoGodel** [comment]
@vyshkouski-godeltech-intelliflo  please, take a look and approve if everything is OK.

**Pull request merged**